File: day-1/solution_part1.lua
local file = io.open("input.txt", "r")
io.input(file)
local file_content = io.read()
io.close(file)

local floor = 0

for i = 1, #file_content do
  if file_content:sub(i, i) == "(" then
    floor = floor + 1
  else
    floor = floor - 1
  end
end

print("The instructions takes santa to floor number " ..floor.. ".");

File: day-1/solution_part1.py
f = open("input.txt", "r").read()

# possible moves are ( and )
# ( = up 1 floor
# ) = down 1 floor

possibleMoves = {
    '(': 1,
    ')': -1
}

floor = 0
for c in f:
    floor += possibleMoves[c]

print('The instructions takes Santa to floor: ' + str(floor) + ' (part 1)')
